pkg/builder.go

```diff
@@ -44,22 +44,27 @@ var (
 
 // Builder interface exposes the method for building a state diff between two blocks
 type Builder interface {
-	BuildStateDiffObject(args Args, params Params) (sd.StateObject, error)
+	BuildStateDiffObject(args sd.Args, params sd.Params) (sd.StateObject, error)
 	BuildStateTrieObject(current *types.Block) (sd.StateObject, error)
 }
 
 type builder struct {
 	stateCache state.Database
+	numWorkers uint
 }
 
 type iterPair struct {
 	older, newer trie.NodeIterator
 }
 
 // NewBuilder is used to create a statediff builder
-func NewBuilder(stateCache state.Database) Builder {
+func NewBuilder(stateCache state.Database, workers uint) Builder {
+	if workers == 0 {
+		workers = 1
```

Comment:
We should add a % 2  guard on this so we always have even

Reply:
Good call - done